src/VisualStudio/CSharp/Impl/Debugging/LocationInfoGetter.cs

```diff
@@ -43,49 +28,93 @@ internal static class LocationInfoGetter
             // If the position does not point to one, try using the first one.
-            if (memberDecl != null &&
-                (memberDecl.Kind() == SyntaxKind.FieldDeclaration || memberDecl.Kind() == SyntaxKind.EventFieldDeclaration))
+            if (memberDeclaration != null &&
+                (memberDeclaration.Kind() == SyntaxKind.FieldDeclaration || memberDeclaration.Kind() == SyntaxKind.EventFieldDeclaration))
             {
-                SeparatedSyntaxList<VariableDeclaratorSyntax> variableDeclarators = ((BaseFieldDeclarationSyntax)memberDecl).Declaration.Variables;
+                SeparatedSyntaxList<VariableDeclaratorSyntax> variableDeclarators = ((BaseFieldDeclarationSyntax)memberDeclaration).Declaration.Variables;
 
                 foreach (var declarator in variableDeclarators)
                 {
                     if (declarator.FullSpan.Contains(token.FullSpan))
                     {
-                        memberDecl = declarator;
+                        fieldDeclarator = declarator;
                         break;
                     }
                 }
 
-                if (memberDecl == null)
+                if (fieldDeclarator == null)
                 {
-                    memberDecl = variableDeclarators.Count > 0 ? variableDeclarators[0] : null;
+                    fieldDeclarator = variableDeclarators.Count > 0 ? variableDeclarators[0] : null;
                 }
             }
 
-            if (memberDecl != null)
+            if (memberDeclaration != null)
             {
-                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-                var memberSymbol = semanticModel.GetDeclaredSymbol(memberDecl, cancellationToken);
+                const char dotTokenText = '.';
+                const string missingInformationPlaceholder = "?";
 
-                if (memberSymbol != null)
+                var builder = s_stringBuilderPool.AllocateAndClear();
+                // simple name
+                var nameToken = fieldDeclarator?.Identifier ?? memberDeclaration.GetNameToken();
+                if (nameToken.IsMissing)
+                {
+                    builder.Append(missingInformationPlaceholder);
+                }
+                else if (nameToken == default(SyntaxToken))
+                {
+                    Debug.Assert(memberDeclaration.Kind() == SyntaxKind.ConversionOperatorDeclaration);
+                    builder.Append((memberDeclaration as ConversionOperatorDeclarationSyntax)?.Type);
+                }
+                else
+                { 
+                    if (memberDeclaration.Kind() == SyntaxKind.DestructorDeclaration)
+                    {
+                        builder.Append('~');
+                    }
+                    builder.Append(nameToken.Text);
+                }
+                // parameter list (if any)
+                builder.Append(memberDeclaration.GetParameterList());
```

Comment:
It would be a reasonable optimization, but I'm not sure it's worth it to muck with the other callers of ```GetParameterList```.